pkg/stolon/stolon.go

```diff
@@ -79,24 +78,28 @@ func (c Clusterdata) Master() DB {
 	return c.Dbs[c.Proxy.Spec.MasterDbUID]
 }
 
-// CheckHealthy returns an error if any of the keepers are marked as unhealthy
+// CheckHealthy returns an error if too many keepers are marked as unhealthy
+// We define healthy as having a healthy master and two other healthy keepers
 func (c Clusterdata) CheckHealthy() error {
 	if reflect.DeepEqual(c.Master(), DB{}) {
 		return errors.New("no healthy master")
 	}
 
-	errors := []string{}
+	healthy := 0
```

Comment:
Can we simplify this slightly by looping through our synchronous standbys?

```
healthyStandbys := 0
for _, standby := range c.SynchronousStandbys() {
  if standby.Status.Healthy {
    healthyStandbys++
  }
}

if healthyStandbys < c.Cluster.Spec.MinSynchronousStandbys {
  error
}
```
?

Reply:
Was looking at https://github.com/gocardless/stolon-pgbouncer/blob/master/pkg/failover/failover.go#L273-L289 for inspiration

Reply:
Discussed in person and came up with a new approach.